src/GUI/Overlay/OverlaySystem.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <Engine/Graphics/Renderer/RendererEnums.h>
+
+#include <unordered_map>
+#include <string>
+#include <memory>
+#include <utility>
+#include <functional>
+
+class Overlay;
+class GameConfig;
+class IOverlayRenderer;
+class PlatformApplication;
+class Renderer;
+struct nk_context;
+
+/** 
+ * @brief An Overlay is a screen built in Immediate Mode GUI ( nuklear ) which is displayed on top of any other ui and it's 
+ * mainly used for debug purposes ( Console, Render Stats, Debug View Buttons, etc... ).
+ * 
+ * The Overlay System is the class responsible to keep track of all the registered Overlays and update them accordingly.
+ */
+class OverlaySystem {
+ public:
+    OverlaySystem(Renderer& render, RendererType rendererType, PlatformApplication &platformApplication);
+    ~OverlaySystem();
+
+    void addOverlay(std::string_view name, std::unique_ptr<Overlay> overlay);
+    void removeOverlay(std::string_view name);
+
+    void update();
+
+ private:
+    std::unordered_map<std::string, std::unique_ptr<Overlay>> _overlays;
```

Comment:
You might need to use Transparent* classes here to do the fixes requested.

Reply:
Seems like everything worked without the need to change it.